tests/tests/proc_macros.rs

```diff
@@ -277,3 +284,23 @@ async fn macro_zero_copy_cow() {
 
 	assert_eq!(result, r#"{"jsonrpc":"2.0","result":"Zero copy params: false, false","id":0}"#);
 }
+
+#[tokio::test]
+async fn multiple_blocking_calls_overlap() {
+	let module = RpcServerImpl.into_rpc();
+
+	let params = RawValue::from_string("[]".into()).ok();
+	// MacOS CI has very limited number of cores, so running on 2 to be safe
+	let futures = std::iter::repeat_with(|| module.call("foo_blocking_call", params.clone())).take(2);
+	let now = Instant::now();
+	let results = futures::future::join_all(futures).await;
+	let elapsed = now.elapsed();
+
+	for result in results {
+		let result = serde_json::from_str::<serde_json::Value>(&result.unwrap()).unwrap();
+		assert_eq!(result["result"], 42);
+	}
+
+	// Each request takes 50ms, added 10ms margin for scheduling
+	assert!(elapsed < Duration::from_millis(60), "Expected less than 60ms, got {:?}", elapsed);
```

Comment:
I'm hard pressed to come up with something better than this. It'll have to make do.